### PR TITLE
SAML Assertion Inline Hook doc

### PR DESCRIPTION
**Pull request by motikagan-okta**
Should be published when SAML Inline Hook functionality is released in EA, currently targeted for 2019.04.0 release.

### CODE REVIEW

File: packages/@okta/vuepress-site/use_cases/inline_hooks/index.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Okta defines several different types of inline hooks. Each type of inline hook m
 | ------------------------------------------------------------------------------------ | ------------------------------------------------          |
 | [Token Inline Hook](/use_cases/inline_hooks/token_hook/token_hook)                   | Customizes tokens returned by Okta API Access Management. |
 | [Import Inline Hook](/use_cases/inline_hooks/import_hook/import_hook)                | Adds custom logic to the user import process.             |
+| [SAML Assertion Inline Hook](/use_cases/inline_hooks/saml_hook/saml_hook)            | Customizes SAML assertions returned by Okta.              |
```
Comment (amoghvasekar-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/index.md]: Is the `type` for this hook, to be used in the management API documented somewhere in this PR? Quick search did not reveal it.

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/index.md]: No, you're right, @amoghvasekar-okta, but I planned to add that to the reference doc for the management API, in the "Supported Inline Hooks" section: https://developer.okta.com/docs/api/resources/inline-hooks/#supported-inline-hook-types
That will be a separate PR for that small addition to that page.

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/index.md]: Have now added this to this pull request.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+  "eventType": "com.okta.saml.tokens.transform",
+  "contentType": "application/json"
+}
+```
+
+## Sample Listing of JSON Payload of Response
+
+```json
+{
+  "error": null,
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: It is better to remove the object with `null` value.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+
+1. Click the **General** tab.
+
+1. In the SAML Settings section, click **Edit**.
+
+1. Click **Show Advanced Settings**.
+
+1. Click **Next** to get to the **Configure SAML** section.
+
+1. In the **Assertion callback** field, select your registered inline hook.
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: "Assertion Inline Hook"

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Thank you, @yuliu-okta. Now fixed.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md (hunk 2 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Removed.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+
+### Specifying Location within the Assertion
+
+You specify the location within the assertion at which to apply your operation using a slash-delimited path.
+
+When performing an `add` op to add a new attribute statement, this will always begin with `/claims/` and be followed by the name of the new attribute you are adding.
+
+When modifying an existing assertions statement, the path could begin with `/subject/`, `/authentication/`, `/conditions/`, or `/claims/`, depending on which part of the assertion you want to modify. You then drill down within the child elements using slash-delimited element names, e.g., `/claims/array/attributeValues/1/value`.
+
+### error
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: We do not support it yet, it should be removed.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+                  "xsi:type": "xs:string"
+                },
+                "value": "barer"
+              }
+            ]
+          }
+        },
+        {
+          "op": "replace",
+          "path": "/conditions/audienceRestriction",
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Let's remove it, we will soon restrict the ability to change audience.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md (hunk 4 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: @yuliu-okta, just to confirm, you're referring to the error object?

Comment (yuliu-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Yes

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Removed.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md (hunk 5 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Removed.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+
+| Property | Description                                                                                                                                                                                                           | Data Type       |
+|----------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|
+| op       | The name of one of the [supported ops](#list-of-supported-ops).                                                                                                                                                       | String          |
+| path     | Location, within the assertion, to apply the operation. See [Specifying Location within the Assertion](##specifying-location-within-the-assertion) below. | String          |
+| value    | Value to set the claim to.                                                                                                                                                                                            | Any JSON object |
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: Hm, would we consider a plain string to be a JSON object? i.e.
`"value": "a string"`;
vs
```
"value": {
   "key": "string in an object";
}
```
both of these are valid, depending on the context.

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+
+#### List of Supported Ops
+
+| Op      | Description                             |
+|---------|-----------------------------------------|
+| add     | Add a new claim to the assertion.       |
+| replace | Modify an existing attribute statement. |
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: `replace` can modify _anything_ in the assertion object, not just attribute statements

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+
+### Specifying Location within the Assertion
+
+You specify the location within the assertion at which to apply your operation using a slash-delimited path.
+
+When performing an `add` op to add a new attribute statement, this will always begin with `/claims/` and be followed by the name of the new attribute you are adding.
+
+When modifying an existing assertions statement, the path could begin with `/subject/`, `/authentication/`, `/conditions/`, or `/claims/`, depending on which part of the assertion you want to modify. You then drill down within the child elements using slash-delimited element names, e.g., `/claims/array/attributeValues/1/value`.
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: You should specify that:
1. This is following JSON Patch conventions
2. Explain that the `/1/` is specifying the index in an array

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+        "foobie": {
+          "attributes": {
+            "NameFormat": "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
+          },
+          "attributeValues": [
+            {
+              "attributes": {
+                "xsi:type": "xs:string"
+              },
+              "value": "doobie"
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: same "foobie"/"doobie" comment

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+        "middle": {
+          "attributes": {
+            "NameFormat": "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
+          },
+          "attributeValues": [
+            {
+              "attributes": {
+                "xsi:type": "xs:string"
+              },
+              "value": "middellllll"
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: nit: change this to something less like me just smashing the keyboard

File: packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+        }
+      ]
+    }
+  }
+}
+```
+
+### data.assertion.lifetime
+
+Specifies the expiration time, in seconds, of the SAML assertion.
```
Comment (danielzeller-okta) [inline packages/@okta/vuepress-site/use_cases/inline_hooks/saml_hook/saml_hook/index.md]: An example here would be great

**Pull request merged**